### PR TITLE
[explorer] Enabling localnet e2e tests

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
This PR introduces new localnet end-to-end tests in the explorer. These tests leverage the `sui-test-validator`, which is a crate we've built that allows us to easily start a local network. These tests are true end-to-end tests, invoking the API and validating the application. This should give us confidence (both on the explorer development side, and also on network/rpc development side) that for any given commit of the `sui` repo, the explorer should work with the network.

A neat side-effect of this approach is that this also works for deployed explorers in other environments. For any network with a faucet, we can run end-to-end tests and validate the deployed explorer works for a specific network. A subset of tests can also run on non-faucet networks.

I looked into two alternatives, static mocks (what we have today), and generated static data.

- **Static Mocks**: This is what we do today, manually curating static data which is used in the UI when run with `pnpm dev:static`. The UI itself is aware of the static data source, and will render a different UI when it is running in static mode. Because of this, these tests aren't really end-to-end tests, and don't validate the same version of the UI that is used for live networks. The manual curation also takes work, and easily becomes out-of-date from the actual API responses. These static UIs also don't actually invoke the API, meaning we don't have confidence that the API calls will actually succeed.

- **Generated Static Data**: This is an approach, where we leverage the `sui-open-rpc` crate to start a local network, put it into a known state, and then capture data from the network into static json files. This approach is similar to the static mock approach, but it avoids network staleness by generating the data from the network. This approach has similar problems to the static data problem, namely not invoking the API and still using static data UI paths. One possibility to solve this is to mock the API responses with the data from the API, but this is complicated to get right, and still doesn't validate that the explorer actually invokes the API correctly, and the separation means it's still possible to introduce changes to the API that break the explorer. This approach also doesn't support more complex chain interactions, like creating and sending transactions to the network. Currently the explorer is a read-only view, but this won't always be the case, so being able to run transactions on the network is critical for our future end-to-end tests. The generated static data is also used currently in the TypeScript SDK to validate the API, but this suffers from the same problems. For this reason, I've talked with Chris, and we've discussed moving the TypeScript SDK tests to leverage a local network as well.
--- 
This adds a new suite of tests that run against localnet. I'm trying to break these tests up more so that they can be written and maintained a little easier. Right now, I'm keeping static tests, and will be slowly moving more over to the localnet tests.

### Running tests

1. Start the `sui-test-validator`. I use `cargo run --bin sui-test-validator`.
2. Start the explorer using `pnpm --filter sui-explorer dev:local`.
3. Open cypress with `pnpm --filter sui-explorer exec cypress open`.

You should now be able to execute the localnet tests!

### CODE REVIEW

**Review by Andrew47 — CHANGES_REQUESTED** [review]
There needs to be more clarity on the value of having these tests specifically against localnet. It adds a requirement that a local network needs to be spun up, which is another thing to go wrong. The original plan proposed by the backend engineers was that we were going to set up tests against the files in `crates/sui-open-rpc/samples`. The backend engineers could then re-generate these files upon breaking changes in the network. Why are we moving away from this plan?

**Comment by Andrew47** [comment]
Having tried to run `cargo run --bin sui-test-validator` locally, turns out the local network breaking is not theoretical:

image

Could a list of requirements for running these tests be provided?

**Comment by todd-mystenlabs** [comment]
Hey folks!

I fully support this change here.  As an entire team, we've taken many hits to productivity when we've unintentionally broken the explorer, wallet, etc.  I want, as closely as possible, for all of us to know that a given change (PR) is in a consistent state across both the Sui core and the other apps that build on it within the repo. @Andrew47 - hopefully that covers your concern there.  It's about the whole team having a stable, known, working main branch at all times.

**Review by Andrew47 — APPROVED** [review]
Thank you very much @Jordan-Mysten for the explanation.

**Comment by 666lcz** [comment]
Not sure why, the `home page` tests pass but the `search` test fails for me
CleanShot 2022-09-02 at 07 57 48

**Review by 666lcz — APPROVED** [review]

File: explorer/client/cypress/localnet.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export async function createLocalnetTasks() {
+    const addressToKeypair = new Map<string, Keypair>();
+
+    return {
+        async faucet() {
+            const keypair = Ed25519Keypair.generate();
+            const address = keypair.getPublicKey().toSuiAddress();
+            addressToKeypair.set(address, keypair);
+            const res = await axios.post<{ ok: boolean }>(
+                'http://localhost:9123/faucet',
```
Comment (666lcz) [inline explorer/client/cypress/localnet.ts:19]: Works for me locally after this change
```suggestion
                'http://127.0.0.1:9123/faucet',
```

**Pull request merged**